osu.Server.Spectator.Tests/Multiplayer/MultiplayerQueueTests.cs

```diff
@@ -0,0 +1,122 @@
+
+            await Hub.JoinRoom(ROOM_ID);
+            await Assert.ThrowsAsync<InvalidStateException>(() => Hub.AddPlaylistItem(new MultiplayerPlaylistItem
+            {
+                BeatmapID = 3333,
+                BeatmapChecksum = "checksum"
+            }));
+        }
+
+        [Fact]
+        public async Task AddCustomizedBeatmapThrows()
+        {
+            Database.Setup(d => d.GetBeatmapChecksumAsync(9999)).ReturnsAsync("correct checksum");
+
+            await Hub.JoinRoom(ROOM_ID);
+            await Assert.ThrowsAsync<InvalidStateException>(() => Hub.AddPlaylistItem(new MultiplayerPlaylistItem
+            {
+                BeatmapID = 9999,
+                BeatmapChecksum = "incorrect checksum",
+            }));
+        }
+
+        [Theory]
+        [InlineData(ILegacyRuleset.MAX_LEGACY_RULESET_ID + 1)]
+        [InlineData(-1)]
+        public async Task AddCustomRulesetThrows(int rulesetID)
+        {
+            await Hub.JoinRoom(ROOM_ID);
+            await Assert.ThrowsAsync<InvalidStateException>(() => Hub.AddPlaylistItem(new MultiplayerPlaylistItem
+            {
+                BeatmapID = 1234,
+                BeatmapChecksum = "checksum",
+                RulesetID = rulesetID
+            }));
+        }
+
+        [Fact]
+        public async Task RoomStartsWithCurrentPlaylistItem()
+        {
+            await Hub.JoinRoom(ROOM_ID);
+
+            using (var usage = Hub.GetRoom(ROOM_ID))
+            {
+                var room = usage.Item;
+                Debug.Assert(room != null);
+
+                Assert.Equal(1, room.Settings.PlaylistItemId);
+            }
+        }
+
+        [Fact]
+        public async Task RoomStartsWithCorrectQueueingMode()
+        {
+            Database.Setup(d => d.GetBeatmapChecksumAsync(3333)).ReturnsAsync("3333");
+            Database.Setup(db => db.GetRoomAsync(ROOM_ID))
+                    .Callback<long>(InitialiseRoom)
+                    .ReturnsAsync(() => new multiplayer_room
+                    {
+                        type = database_match_type.head_to_head,
+                        ends_at = DateTimeOffset.Now.AddMinutes(5),
+                        user_id = USER_ID,
+                        queue_mode = database_queue_mode.free_for_all
+                    });
+
+            await Hub.JoinRoom(ROOM_ID);
+
+            await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
+            {
+                BeatmapID = 3333,
+                BeatmapChecksum = "3333"
+            });
+
+            using (var usage = Hub.GetRoom(ROOM_ID))
+            {
+                var room = usage.Item;
+                Debug.Assert(room != null);
+
+                Assert.Equal(2, (await Database.Object.GetAllPlaylistItems(ROOM_ID)).Length);
+            }
+        }
+
+        [Fact]
+        public async Task JoiningRoomInvokesPlaylistItemAddedCallbackForEveryItem()
```

Comment:
This test is failing for me locally. Adding an `await Hub.RequestAllPlaylistItems();` fixes, but honestly I'd more expect it to pass as written (as per [my client-side pull comment](https://github.com/ppy/osu/pull/15640#discussion_r749626537)).

Reply:
Should be fixed now because the method doesn't exist anymore.